hw/top_earlgrey/syn/asic.constraints.sdc

```diff
@@ -91,41 +85,65 @@ set_output_delay ${IO_DIV2_OUT_DEL} [get_ports IOC*] -clock IO_DIV2_CLK
 set_output_delay ${IO_DIV2_OUT_DEL} [get_ports IOR*] -clock IO_DIV2_CLK
 
 # IO_DIV4_CLK
-set IO_DIV4_IN_DEL    [expr ${IO_IN_DEL_FRACTION} * ${IO_TCK} * 4.0]
-set IO_DIV4_OUT_DEL   [expr ${IO_OUT_DEL_FRACTION} * ${IO_TCK} * 4.0]
+set IO_DIV4_IN_DEL    [expr ${IO_IN_DEL_FRACTION} * ${IO_TCK_PERIOD} * 4.0]
+set IO_DIV4_OUT_DEL   [expr ${IO_OUT_DEL_FRACTION} * ${IO_TCK_PERIOD} * 4.0]
 
-set_input_delay ${IO_DIV4_IN_DEL}   [get_ports IOA*] -clock IO_DIV4_CLK
-set_input_delay ${IO_DIV4_IN_DEL}   [get_ports IOB*] -clock IO_DIV4_CLK
-set_input_delay ${IO_DIV4_IN_DEL}   [get_ports IOC*] -clock IO_DIV4_CLK
-set_input_delay ${IO_DIV4_IN_DEL}   [get_ports IOR*] -clock IO_DIV4_CLK
+set_input_delay ${IO_DIV4_IN_DEL}   [get_ports IOA*] -clock IO_DIV4_CLK -add_delay
+set_input_delay ${IO_DIV4_IN_DEL}   [get_ports IOB*] -clock IO_DIV4_CLK -add_delay
+set_input_delay ${IO_DIV4_IN_DEL}   [get_ports IOC*] -clock IO_DIV4_CLK -add_delay
+set_input_delay ${IO_DIV4_IN_DEL}   [get_ports IOR*] -clock IO_DIV4_CLK -add_delay
 
-set_output_delay ${IO_DIV4_OUT_DEL} [get_ports IOA*] -clock IO_DIV4_CLK
-set_output_delay ${IO_DIV4_OUT_DEL} [get_ports IOB*] -clock IO_DIV4_CLK
-set_output_delay ${IO_DIV4_OUT_DEL} [get_ports IOC*] -clock IO_DIV4_CLK
-set_output_delay ${IO_DIV4_OUT_DEL} [get_ports IOR*] -clock IO_DIV4_CLK
+set_output_delay ${IO_DIV4_OUT_DEL} [get_ports IOA*] -clock IO_DIV4_CLK -add_delay
+set_output_delay ${IO_DIV4_OUT_DEL} [get_ports IOB*] -clock IO_DIV4_CLK -add_delay
+set_output_delay ${IO_DIV4_OUT_DEL} [get_ports IOC*] -clock IO_DIV4_CLK -add_delay
+set_output_delay ${IO_DIV4_OUT_DEL} [get_ports IOR*] -clock IO_DIV4_CLK -add_delay
 
 #####################
-# AON clk (300kHz)  #
+# AON clk           #
 #####################
-set AON_CLK_PIN ast_wrapper/i_ast/i_prim_clock_buf_aon/*/u_size_only_buf/${DRIVING_CELL_PIN}
-set AON_TCK 3333.0
-set_ideal_network [get_pins ${AON_CLK_PIN}]
+set AON_CLK_PIN ast_wrapper/i_ast/i_aon_clk/i_aon_osc/aon_clk_o
+# target is 200KHz, overconstrain by factor 
+set AON_TCK_TARGET_PERIOD 5000.0
+set AON_TCK_PERIOD [expr $AON_TCK_TARGET_PERIOD*$FACTOR]
+#set_ideal_network [get_pins ${AON_CLK_PIN}]
 
-create_clock -name AON_CLK -perio ${AON_TCK} [get_pins ${AON_CLK_PIN}]
+create_clock -name AON_CLK -period ${AON_TCK_PERIOD} [get_pins ${AON_CLK_PIN}]
 set_clock_uncertainty ${SETUP_CLOCK_UNCERTAINTY} [get_clocks AON_CLK]
 
 #####################
 # JTAG clock        #
 #####################
-# TODO: set up constraints for JTAG. this may need additional views with set_case_analysis
+# TODO: set up constraints for JTAG. 
+set JTAG_CLK_PIN IOR3
+# target is 20MHz, overconstrain by factor 
+set JTAG_TCK_TARGET_PERIOD 50
+set JTAG_TCK_PERIOD [expr $JTAG_TCK_TARGET_PERIOD*$FACTOR]
+
+create_clock -name JTAG_TCK -period $JTAG_TCK_PERIOD [get_ports $JTAG_CLK_PIN]
```

Comment:
@arnonsha just a quick question here, any thoughts on how we want to handle this eventually when jtag clock is muxed with other functional pins? I know we discussed this on the call also.

Reply:
I thought that was exactly what set_case_analysis should handle?

Reply:
Actually the set_case_analysis will prevent timing checks on all paths beside jtag...

In the muxed IO case when set_case_analysis is not uses DC/ICC will see the paths from pad to any FF clk pin (in the jtag tap module), and consider this as clock. It will also check any data path between the pad and non JTAG related logic (an the other way around).

So the bottom line: no need to do anything. The tools will check all clock/non clock paths from logc to pad and from pad to logic.

Reply:
sounds good.  Hopefully by setting jtag clock async to all the internal clocks, even if the pad touches the D pin of some internal FF, it should expend any effort to try to balance?

Reply:
It shouldn't as this is data path. 
if worst comes to worst there are commands to stop the clock propagation to certain paths.